apps/src/netsim/NetSimTable.js

```diff
@@ -31,17 +31,24 @@ var DEFAULT_POLLING_DELAY_MS = 5000;
  * Wraps the app storage table API in an object with local
  * cacheing and callbacks, which provides a notification API to the rest
  * of the NetSim code.
+ * @param {!string} shardID - The shard ID specific to this class' NetSim instance.
  * @param {!string} tableName - The name of the remote storage table to wrap.
  * @constructor
+ * @throws {Error} if wrong number of arguments are provided.
  */
-var NetSimTable = module.exports = function (tableName) {
+var NetSimTable = module.exports = function (shardID, tableName) {
+  // Require both shardID and tableName to be provided
+  if (!shardID || !tableName) {
+    throw new Error('NetSimTable must be constructed with both arguments. ' +
```

Comment:
I'm a little confused; it looks like you're commenting on the JavaScript exception in client code, which does not cause any sort of HTTP request.  Are you saying the server have a custom handler for the sort of malformed URL that would happen if we _didn't_ catch this case?